### ISSUE TITLE
Action blocklist_de (api key clarification)

### ISSUE THREAD

**Issue by mitchellkrogza** [fail2ban/fail2ban#1940]
_**FIRSTLY:**_

Documentation in `jail.conf` specifies for blocklist.de that a `jail.d/blocklist_de.local`is created containing 

```
[Init]
blocklist_de_apikey = {api key from registration]
```

I have tried this but the `jail.d/blocklist_de.local` is not detected, how does one tell fail2ban to include the `jail.d/*` config files when loading?

**_SECONDLY:_**

Can you clarify that the api key is actually wrapped beginning with `{` and ending with `]` so beginning with a curly bracket and ending with a square bracket?

I have managed to get fail2ban to load and not fail by adding this `blocklist_de_apikey = {x55x55x55x]` into my jail.local file but nothing is ever being reported to blocklist.de even though I get emails when a ban has occured. 

I have tried various combinations to like `blocklist_de_apikey = {x55x55x55x}` and `blocklist_de_apikey = x55x55x55x` but not reports are ever being sent to blocklist.de but reporting to badips within the same jails and emailing to me all works.

I would really like to get this working properly but have now tried various combinations spanning a few days and of course have to always wait until a ban occurs to find out if my changes worked. 

Really would appreciate some clarity on this. Many thanks.

**Comment by mitchellkrogza** [comment]
For anyone interested in how to get this working. First do NOT use the method specifying to create the file `jail.d/blocklist_de.local` it does not work.

Manually modify this line below in your jail.local as follows

`action_blocklist_de  = blocklist_de[email="%(sender)s", service=%(filter)s, apikey="xxxxxxxxx"]`

Where xxxxxxxxx is your actual API Key. Simply the API key string they issued you when you added your server without any curly or square brackets.

Then in your jail(s) do the following, this example of mine shows how to make one jail use 3 actions, sending the notification to me, reporting it to badips.com and reporting it to blocklist.de

```
[ssh]
enabled  = true
port     = ssh
filter   = sshd
logpath  = /var/log/auth.*
maxretry = 6
action = %(action_mwl)s
              badips[category="sshd", key="my-api-key-for-bad-ips-dot-com"]
              blocklist_de
```

Restart Fail2ban and all actions will work.

You can also configure it like this if you want.

```
[ssh]
enabled  = true
port     = ssh
filter   = sshd
logpath  = /var/log/auth.*
maxretry = 6
action = %(action_mwl)s
              badips[category="sshd", key="my-api-key-for-badips-dot-com"]
              blocklist_de[email="%(sender)s", service=%(filter)s, apikey="my-api-key"]
```

and if you have a [blacklist filter like my permaban/blacklist filter her](https://github.com/mitchellkrogza/Fail2Ban-Blacklist-JAIL-for-Repeat-Offenders-with-Perma-Extended-Banning)e you can customize it as follows:

```
[blacklist]
enabled = true
logpath  = /var/log/fail2ban.*
filter = blacklist
banaction = blacklist
action = %(action_mwl)s
              badips[category="badbots", key="my-api-key-for-bad-ips-dot-com"]
              blocklist_de[email="%(sender)s", service=badbots, apikey="my-api-key"]
bantime  = 31536000   ; 1 year
findtime = 31536000   ; 1 year
maxretry = 10
```

I hope this helps somebody.

**Comment by sebres** [comment]
Documentation fixed in e07a8cda07c8b3c3e9ab6ed304ac547fcbff986c now.

BTW
1. Files `jail.d/my-jail.conf` resp. `jail.d/my-jail.local` will be loaded only if jail `my-jail` will be enabled in `jail.local`. 
2. The section `[Init]` used only in actions (not in jails), thus not correct for `jail.d/my-jail.local`.
3. But it will work if all expected init-parameters specified in `action.d/blocklist_de.conf` will be overwritten in `action.d/blocklist_de.local` (without supplying of all this parameters in jail, like substitution of `%(action_blocklist_de)s`).
